ExchangeSharp/API/Exchanges/Bitfinex/ExchangeBitfinexAPI.cs

```diff
@@ -223,6 +223,69 @@ protected override IWebSocket OnGetTickersWebSocket(Action<IReadOnlyCollection<K
             });
         }
 
+        protected override IWebSocket OnGetTradesWebSocket(Action<KeyValuePair<string, ExchangeTrade>> callback, params string[] symbols)
+        {
+            if (callback == null)
+            {
+                return null;
+            }
+            Dictionary<int, string> channelIdToSymbol = new Dictionary<int, string>();
+            return ConnectWebSocket("/2", (_socket , msg) => //use websocket V2 (beta, but millisecond timestamp)
+            {
+                JToken token = JToken.Parse(msg.ToStringFromUTF8());
```

Comment:
Depending on how heavy the traffic is on this websocket, you can gain 20% perf improvement deserializing JSON to a predefined object.